deploy/7-deploy-pendle-oracle.ts

```diff
@@ -4,49 +4,119 @@ import { HardhatRuntimeEnvironment } from "hardhat/types";
+    args,
+    proxy: {
+      owner: proxyOwnerAddress,
+      proxyContract: "OptimizedTransparentUpgradeableProxy",
+      viaAdminContract: {
+        name: "DefaultProxyAdmin",
+        artifact: defaultProxyAdmin,
+      },
+    },
+    skipIfAlreadyDeployed: true,
+  });
+};
+
 const func: DeployFunction = async ({
   getNamedAccounts,
   deployments,
   network,
   artifacts,
 }: HardhatRuntimeEnvironment) => {
-  const { deploy } = deployments;
   const { deployer } = await getNamedAccounts();
 
   const oracle = await ethers.getContract("ResilientOracle");
   const proxyOwnerAddress = network.live ? ADDRESSES[network.name].timelock : deployer;
-
   const defaultProxyAdmin = await artifacts.readArtifact(
     "hardhat-deploy/solc_0.8/openzeppelin/proxy/transparent/ProxyAdmin.sol:ProxyAdmin",
   );
 
-  const { PTweETH_26DEC2024, PTweETH_26DEC2024_Market, PTOracle, WETH } = ADDRESSES[network.name];
+  const addresses = ADDRESSES[network.name];
+  const fallbackAddress = "0x0000000000000000000000000000000000000001";
 
-  const ptOracleAddress = PTOracle || (await ethers.getContract("MockPendlePtOracle")).address;
+  const deployOracles = async (oracleConfig: OracleConfig) => {
+    for (const { name, market, token, collateral, customOracle } of oracleConfig) {
+      const ptOracleAddress = customOracle || (await ethers.getContract("MockPendlePtOracle")).address;
+      await deployPendleOracle(
+        deployments,
+        deployer,
+        name,
+        [market || fallbackAddress, ptOracleAddress, token, collateral, oracle.address, 1800],
+        proxyOwnerAddress,
+        defaultProxyAdmin,
+      );
+    }
+  };
 
-  await deploy("PendleOracle-PT-weETH-26DEC2024", {
-    contract: "PendleOracle",
-    from: deployer,
-    log: true,
-    deterministicDeployment: false,
-    args: [
-      PTweETH_26DEC2024_Market || "0x0000000000000000000000000000000000000001",
-      ptOracleAddress,
-      PTweETH_26DEC2024,
-      WETH,
-      oracle.address,
-      1800,
-    ],
-    proxy: {
-      owner: proxyOwnerAddress,
-      proxyContract: "OptimizedTransparentUpgradeableProxy",
-      viaAdminContract: {
-        name: "DefaultProxyAdmin",
-        artifact: defaultProxyAdmin,
-      },
+  const oracleConfig: OracleConfig = [
+    {
+      name: "PendleOracle-PT-weETH-26DEC2024",
+      market: addresses.PTweETH_26DEC2024_Market,
+      token: addresses.PTweETH_26DEC2024,
+      collateral: addresses.WETH,
+      customOracle: addresses.PTOracle || (await ethers.getContract("MockPendleOracle")).address,
     },
-    skipIfAlreadyDeployed: true,
-  });
+    {
+      name: "PendleOracle_PT_USDe_27MAR2025",
+      market: addresses.PTUSDe_27MAR2025_Market,
+      token: addresses.PTUSDe_27MAR2025,
+      collateral: addresses.USDe,
+      customOracle:
+        network.name === "sepolia"
+          ? (await ethers.getContract("MockPendleOracle_PT_USDe_27MAR2025")).address
+          : addresses.PTOracle,
```

Comment:
`PTOracle` for Ethereum mainnet is set to `0x66a1096C6366b2529274dF4f5D8247827fe4CEA8`, but we should use the new PendleOracle here (`0x9a9fa8338dd5e5b2188006f1cd2ef26d921650c2`). So, I would add a new entry in the deploymentConfig.ts file for the new PendleOracle on Ethereum, and I would it here